linkerd/proxy/http/src/insert.rs

```diff
@@ -26,46 +44,83 @@ pub struct FnLazy<F>(F);
 #[derive(Clone, Debug)]
 pub struct ValLazy<V>(V);
 
-/// Wraps an HTTP `Service` so that a `P`-typed `Param` is cloned into each
-/// request's extensions.
+#[pin_project::pin_project]
 #[derive(Debug)]
```

Comment:
nit: this `Debug` impl will require `V: Debug` and `B: Debug`, because they're parameters of the type...but they aren't needed to implement `Debug`. we _probably_ should replace this with a manual impl (if we care)